### DISCUSSION TITLE
Submodule link only works within GitHub

### DISCUSSION THREAD

**Discussion by carsakiller** [community/community#49626] [Repositories]
### Select Topic Area

Bug

### Body

Submodules in your repository [appear as a link](https://github.com/LuaLS/LLS-Addons/tree/main/addons/busted) that, when clicked, will take you to the repository for that submodule on GitHub. However, if your [submodule lives on GitLab](https://github.com/LuaLS/LLS-Addons/tree/main/addons/cc-tweaked) or some other service, the link still tries to find the project on GitHub.

I understand why GitHub would simply link to itself, but I think it severely harms collaboration, as well as safety. If someone wants to see what is in the submodule, they cannot easily do so, they would have to manually dig through `.gitsubmodules` to find the correct link. It would also make it easier for malicious code to sneak in, as it is more difficult to view what is in the submodule.

It seems that `.gitmodules` is already being parsed to find these links to GitHub, I just ask that support for other domains is added as well.

**Comment by rdipardo** [comment]
> However, if your [submodule lives on GitLab][0] or some other service, the link still tries to find the project on GitHub.

There are even cases when *no link* is generated *at all*; for example, if the submodule's name and path attributes differ from each other. [This configuration][1] generates the plain text `Interface @ e76c3be`:

~~~gitconfig
[submodule "DelphiPluginTemplate"]
  path = Interface
  url = # . . .
~~~

Yet changing either the name or path to match creates the circular self-referencing link that @carsakiller described, i.e.,

~~~gitconfig
[submodule "Interface"]
  path = Interface
  url = # . . .
~~~

![circular-sub-project-link-1](https://github.com/community/community/assets/59004801/6b0c254e-a28b-423a-a8fe-442ff3b550cd)

or,

~~~gitconfig
[submodule "DelphiPluginTemplate"]
  path = DelphiPluginTemplate
  url = # . . .
~~~

![circular-subproject-link-2](https://github.com/community/community/assets/59004801/ed3ff493-002d-4d17-871a-f8aa7fae1ec1)

[0]: https://github.com/LuaLS/LLS-Addons/tree/main/addons/cc-tweaked
[1]: https://github.com/rdipardo/nppQrCode/blob/develop/.gitmodules

**Comment by rdipardo** [comment]
[Diff views] are interestingly unaffected by this limitation, e.g.,

![diff-view-with-working-link-to-subproject](https://github.com/community/community/assets/59004801/04fb4d93-67bd-470e-ad73-e292d2d6e860)

[Diff views]: https://github.com/rdipardo/nppQrCode/commit/ee57ed54d0ea87b5fa821396e0e6f45ff14b2384#diff-c26b3ed4ce70e202816608a9ca6cc14da9ed4a5cf14c7c758be0e2c20344ed82
